### PR TITLE
Improving project

### PR DESCRIPTION
**Pull request by dennybiasiolli**
## What did you do?

Updated projects who were using node dependencies.

- husky: removing unused sample script

- database: removing unused husky dependency and script

- services: removing unused husky dependency and script

- ui: removing unused husky dependency and script

- src: fixing prettier errors with `prettier --write "src/**/*.{js,vue,scss}"`

- ui: upgrading serialize-javascript and yargs-parser

- ui: upgrading lit-element and lit-html, are they used??

- ui: upgrading babel-jest dependency

- upgrading amplitude-js to latest version

- running `yarn lint`

- ui: upgrading vue-plugin-load-script and @vue/test-utils

- services,ui: requiring node engine >=14.x

- upgrading dependencies to last version using `yarn upgrade-interactive --latest`

    - node-sass@6.0.1
    - sass-loader@^10 https://stackoverflow.com/a/66087132/3963422
    - webpack@5.61.0

- upgrading vue-cli dependencies with `npx vue upgrade`

    ```
    Upgrading @vue/cli-plugin-babel from 3.12.1 to 4.5.15

    ✨  Done in 6.84s.
    ✔  Successfully invoked migrator for plugin: @vue/cli-plugin-babel
     babel  core-js has been upgraded from v2 to v3.
    If you have any custom polyfills defined in babel.config.js, please be aware their names may have been changed.
    For more complete changelog, see https://github.com/zloirock/core-js/blob/master/CHANGELOG.md#300---20190319
    ```

- yarn: removing all `.yarnrc` and `yarn*.cjs` files

    Not useful and/or used in the project, it's better to use the official
    yarn package installed in the system.

    https://yarnpkg.com/configuration/yarnrc/#yarnPath

- services: upgrading jsdom dependency to latest version

    No breaking changes: https://github.com/jsdom/jsdom/blob/master/Changelog.md

- services: upgrading dotenv dependency to latest version

    No breaking changes: https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md

- services: upgrading axios to latest version to solve high level audit warnings

    - Server-Side Request Forgery in Axios
        solved in axios>=0.21.1
        https://www.npmjs.com/advisories/1002990

    - Incorrect Comparison in axios
        solved in axios>=0.21.2
        https://www.npmjs.com/advisories/1002477

    No breaking changes: https://github.com/axios/axios/blob/master/CHANGELOG.md

- ui: upgrading axios to latest version to solve high level audit warnings

    - Server-Side Request Forgery in Axios
        solved in axios>=0.21.1
        https://www.npmjs.com/advisories/1002990

    - Incorrect Comparison in axios
        solved in axios>=0.21.2
        https://www.npmjs.com/advisories/1002477

    No breaking changes: https://github.com/axios/axios/blob/master/CHANGELOG.md

- services: upgrading outdated dependencies with `yarn upgrade`

- ui: upgrading outdated dependencies with `yarn upgrade`

## Why did you do it?

Keeping up-to-date all dependencies and removing error/warnings with deprecated/critical dependencies.

## How have you tested it?

I've tested it running all test/build/lint scripts inside package.json files.

## Were docs updated if needed?

- [ ] No
- [X] Yes
- [ ] N/A

#### Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new console logs
- [ ] Fixes entire issue
- [ ] Partial fix for issue

### CODE REVIEW

File: database/package.json [hunk 1]
```diff
@@ -6,10 +6,6 @@
   "license": "MIT",
   "scripts": {
     "serve": "docker stop localdev-db; docker rm localdev-db; docker run --detach --rm -p 8888:5984 --name localdev-db five-fifths-voter/db",
-    "build": "docker build -t five-fifths-voter/db .",
-    "postinstall": "cd .. && husky install"
-  },
-  "devDependencies": {
-    "husky": "^7.0.4"
```
Comment (davidnixon) [inline database/package.json:13]: 😄 So we just recently added this as a placeholder to enable some real work in precommits.
#285 
Can you revert this change. I still need to open a new issue asking for at least a lint check in precommit.

**Comment by davidnixon** [comment]
> yarn: removing all .yarnrc and yarn*.cjs files
> 
> Not useful and/or used in the project, it's better to use the official
> yarn package installed in the system.
> 
> https://yarnpkg.com/configuration/yarnrc/#yarnPath

Currently the project is running with yarn 1.x and not 2.x. These files are used to explicitly set the yarn version to yarn-1.22.11.  

I would like to upgrade to yarn 2 and use the monorepo support but I have not opened and issue for that either.

**Comment by davidnixon** [comment]
I don't understand why prettier made changes. When I run the lint check locally I see no issues.
> localhost:~/projects/Five-Fifths-Voter/ui$ yarn ci-check
> yarn run v1.22.11
> $ yarn format:diff
> $ prettier --list-different "src/**/*.{js,vue,scss}"
> ✨  Done in 3.31s.

checking further ...

File: ui/src/components/Absentee/Absentee.vue [hunk 2]
```diff
@@ -38,7 +38,7 @@ export default {
     this.$amplitude
       .getInstance()
       .logEvent('Page Visit', { page: 'Journey Page', component: 'Absentee' });
-  }
+  },
```
Comment (davidnixon) [inline ui/src/components/Absentee/Absentee.vue:41]: When I add this change locally and then run ci-check ...
> localhost:~/projects/Five-Fifths-Voter/ui$ yarn ci-check   
> yarn run v1.22.11
> $ yarn format:diff
> $ prettier --list-different "src/**/*.{js,vue,scss}"
> src/components/Absentee/Absentee.vue
> error Command failed with exit code 1.
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> error Command failed with exit code 1.
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

If I run prettier against the file it removes the trailing comma.

Somehow your prettier is using
```js
"trailingComma": "es5",
```
but it's not in the package.json. Is it possible that you have a prettier config file that is adding this?
https://prettier.io/docs/en/configuration.html

**Review by davidnixon — CHANGES_REQUESTED** [review]
I think we should revery the husky, yarn and prettier changes. Upgrades look good but still verifying those.

@GeraldMit what do you think of allowing node 14+ instead of just 14. We deploy with 14 so I think we should ensure developers are using that but I can see how running nvm use for this project can be annoying.

**Comment by davidnixon** [comment]
Hmmm the prettier config issue is strange and I had not noticed before some inconsistencies.
with a clean repo from the ui dir

> localhost:~/projects/Five-Fifths-Voter/ui$ yarn ci-check                                      
> yarn run v1.22.11
> $ yarn format:diff
> $ prettier --list-different "src/**/*.{js,vue,scss}"
> ✨  Done in 2.70s

But if I run prettier outside of yarn

> localhost:~/projects/Five-Fifths-Voter/ui$ prettier --list-different "src/**/*.{js,vue,scss}"      
> src/components/Absentee/Absentee.vue
> src/components/AppHeader/AppHeader.vue
> src/components/BallotReturn/BallotReturn.vue
> src/components/Countdown/Countdown.vue
> src/components/EarlyVoting/EarlyVoting.vue
> src/components/Hero/Hero.vue
> src/components/MainContent/MainContent.vue
> src/components/Maps/constants/mapSettings.js
> src/components/Maps/GoogleMap.vue
> src/components/Maps/GoogleMapLoader.vue
> src/components/Maps/GoogleMapMarker.vue
> src/components/MarkDown/MarkDown.vue
> src/components/PageLayout/PageLayout.vue
> src/components/Register/Register.vue
> src/components/TwitterChatter/TwitterChatter.vue
> src/components/VoteNow/VoteNow.vue
> src/main.js
> src/plugins/i18n.js
> src/router.js
> src/views/ConnectPage/ConnectPage.vue
> src/views/JourneyPage/JourneyPage.vue
> src/views/LandingPage/LandingPage.vue
> src/views/NotFoundPage/NotFoundPage.vue
> src/views/RatePage/RatePage.vue
> src/views/SupportPage/SupportPage.vue
> src/views/WhyVotePage/WhyVotePage.vue

I'm not sure what is happening here but checking ...

**Comment by dennybiasiolli** [comment]
> Hmmm the prettier config issue is strange and I had not noticed before some inconsistencies. with a clean repo from the ui dir
> 
> > localhost:~/projects/Five-Fifths-Voter/ui$ yarn ci-check
> > ...
> 
> But if I run prettier outside of yarn
> 
> ...
> I'm not sure what is happening here but checking ...

Yarn executes the local version of prettier (inside node_modules) if present, running the `prettier` command will use the globally installed version. Maybe they are different, this is the reason I never use globally installed packages.

For other changes, I'll make the required changes tomorrow 😉

File: database/package.json (hunk 1 above)
Comment (dennybiasiolli) [reply database/package.json]: Ok, husky restored removing removal commits, I will inspect other changes tomorrow

**Comment by dennybiasiolli** [comment]
> Currently the project is running with yarn 1.x and not 2.x. These files are used to explicitly set the yarn version to yarn-1.22.11.

Yes, but it's strange that in the repo there are four copies of `yarn-1.22.11.cjs` file, 4.9MB each.
I solved with a commit referencing to the root `yarn-1.22.11.cjs` file instead of duplicating each one.
We can also update to `yarn-1.22.11.cjs` launching `yarn set version 1` in the root directory.

File: ui/src/components/Absentee/Absentee.vue (hunk 2 above)
Comment (dennybiasiolli) [reply ui/src/components/Absentee/Absentee.vue:41]: I think that `"trailingComma": "es5",` is the default behaviour for prettier: https://prettier.io/docs/en/options.html#trailing-commas

Comment (dennybiasiolli) [reply ui/src/components/Absentee/Absentee.vue:41]: Also, before running `yarn ci-check` make sure to have updated all yarn dependencies in the branch with `yarn install` or simply `yarn`

Comment (davidnixon) [reply ui/src/components/Absentee/Absentee.vue:41]: You are exactly right. I was taking a shortcut and not running yarn from your PR. The `trailingComma` did indeed change in prettier 2.0.

**Review by davidnixon — CHANGES_REQUESTED** [review]
This file was added after you made most of your changes `src/registerServiceWorker.js` can run `ci-format` and add that file to your change set. (Sorry it took me so long to get to this PR.)

Otherwise I think this is ready to merge. Thanks for your patience on this one.

**Review by davidnixon — APPROVED** [review]
